src/mlpack/core/data/split_data.hpp

```diff
@@ -224,6 +228,122 @@ Split(const arma::Mat<T>& input,
                          std::move(testData));
 }
 
+template <class fieldType,
```

Comment:
```suggestion
template <typename FieldType,
```
Two things---I'd suggest using `typename` instead of `class` to match the rest of the codebase (there is no real difference between those two things), and also be sure to capitalize the template parameters (also for consistency). :+1: